front/src/components/AddReservation.tsx

```diff
@@ -0,0 +1,305 @@
+    const [selectedQuantity, setSelectedQuantity] = useState(0);
+    const [disabledQuantity, setDisabledQuantity] = useState(true);
+    const [options, setOptions] = useState([0]); 
+
+    // Component to display a toast message when non-logged user try to add a product to cart
+    const LoginButton = () => {
+        return (
+            <div className="items-center text-center font-bold">
+                Vous devez être connecté pour ajouter un produit à votre panier de réservation.
+                <button 
+                    className="bg-orange-600 hover:bg-orange-700 text-white p-2 rounded-md mt-4"
+                    onClick={() => navigate("/connect")}
+                >
+                    Se connecter / S'inscrire
+                </button>
+            </div>
+        );
+    }
+
+    // Details of a product already rented
+    const [productReserved] = useLazyQuery(GET_DETAILS_OF_ONE_PRODUCT_RESERVED);
+
+    // User cart
+    const GetUserCart = useQuery(GET_USER_CART, {
+        variables: { getCartReservationOfUserId: userId },
+        skip: userId === "",
+    });
+
+    const [createCart] = useMutation(CREATE_CART);
+    const [updateCart] = useMutation(UPDATE_CART);
+
+    const p = async() => {
+        if(date && Array.isArray(date) && date[0] && date[1]) {
+            let { data } = await productReserved({
+                variables: { 
+                    getProductReservedInput: {
+                        product_id: productId,
+                        start_at: date?date[0].toLocaleString("en-US", {timeZone: "Europe/Paris"}):null,
+                        end_at: date?date[1].toLocaleString("en-US", {timeZone: "Europe/Paris"}):null,
+                    },
+                },
+            });
+
+            let stockAvailable = 0;
+            let reservedQuantity = 0;
+
+            const p = data?.getDetailsOfOneProductReserved.map((product:any) => {
+                console.log("MAP : "+product.reservationsDetails[0].quantity)
+                return reservedQuantity += product.reservationsDetails[0].quantity;
+            });
+            
+            stockAvailable = stock - reservedQuantity;
+            stockAvailable = stockAvailable < 0 ? 0 : stockAvailable;
+
+            // If there is a quantity and a date range is selected then we update options values with quantity
+            if (stockAvailable > 0 && typeof date !== 'undefined') {
+                const formSelectOptions = [];
+                for (let i = 1; i <= stockAvailable; i++) {
+                    formSelectOptions.push(i);
+                }
+                setOptions(formSelectOptions);
+                setDisabledQuantity(false);
+            } else {
+                initForm();
+            }
+
+            // Close the calendar
+            if (openModal === true) setOpenModal(true);
+            else setOpenModal(false);
+        }
+    }
+
+    useEffect(() => {
+        p(); 
+    }, [date, selectedQuantity]);
```

Comment:
tu peux désactiver ce warning parce qu'il a pas l'air très pertinent en rajoutant :  


```suggestion
        // eslint-disable-next-line react-hooks/exhaustive-deps
```
après la l 119 avec le }